package.json

```diff
@@ -110,7 +115,6 @@
     "fs-extra": "9.1.0",
     "fuse.js": "6.4.6",
     "globby": "11.0.2",
-    "gulp": "4.0.2",
     "gulp-connect": "5.7.0",
```

Comment:
Can `gulp-connect` be deleted?

Reply:
`gulp-connect` is the library we use for our dev / test server. It's built on top of the more bare-bones `connect` library and provides us with a way to launch a webserver with custom middleware written with [ExpressJS](https://expressjs.com/en/guide/using-middleware.html). We do not use it with `gulp`, and instead, call it via its nodeJS API:

https://github.com/ampproject/amphtml/blob/124d4106de4ddcc0abde76485df8139eb93cf15d/build-system/tasks/serve.js#L101-L118

I set out to replace `gulp-connect` with the `connect` library, but found that I'd basically have to copy or re-implement [all this code](https://github.com/avevlad/gulp-connect/blob/d306eb32c362189e24d2927da712abd589796e42/src/index.coffee#L1-L209) to create a similar interface. I briefly looked for other alternatives that support our Express middleware, but couldn't find one. This is when I moved on.

Happy to revisit this if you're able to find an alternative that doesn't need a substantial reimplementation of middleware logic. (There's got to be something out there.)

WDYT?

Reply:
Totally fine to add a TODO / tracking issue and replace later on!

Reply:
On re-looking at the code for `gulp-connect`, I noticed that it [does not list](https://github.com/avevlad/gulp-connect/blob/d306eb32c362189e24d2927da712abd589796e42/package.json#L46-L56) `gulp` as a dependency. As a result, AMP [no longer depends on](https://github.com/ampproject/amphtml/issues/32585#issuecomment-802435445) `gulp` directly or indirectly. So I'm not sure it's worth the effort to get rid of `gulp-connect` just because of its name 😃